### DISCUSSION TITLE
Code coverage reports with ClearScript v8

### DISCUSSION THREAD

**Discussion by EinmalIM** [ClearFoundry/ClearScript#448] [Q&A]
We are running JavaScipts with the ClearScript v8 engine.

Those scripts export functionality which is imported and tested by other scripts.

Is there a way to get a code coverage report when executing the test scripts?

**Comment by ClearScriptLib** [comment]
Hi @EinmalIM.

>Is there a way to get a code coverage report when executing the test scripts?

Unfortunately, we don't have a definitive answer for you. There are popular JavaScript coverage tools, such as [Istanbul.js](https://istanbul.js.org/), but it isn't clear how much they rely on Node.js infrastructure.

As you probably know, there's an amazing variety of JavaScript tools and libraries out there, but it seems like most are designed for a web or Node.js environment rather than an application scripting environment based on bare standard JavaScript.

One thing you might consider looking into is the built-in coverage tools in Chromium-based browsers such as Chrome or Edge. ClearScript supports V8's debugging interface, so, in theory, it should be possible to attach such a browser to your application and use all the available debugging tools.

Good luck!
